schema/gen/go/adjunctCfg.go

```diff
@@ -74,13 +74,58 @@ func (cfg *AdjunctCfg) MaybeUsesPtr(t schema.Type) bool {
 	if x, ok := cfg.maybeUsesPtr[t.Name()]; ok {
 		return x
 	}
-	// FUTURE: we could make this default vary based on sizeof the type.
-	//  It's generally true that for scalars it should be false by default; and that's easy to do.
-	//   It would actually *remove* special cases from the prelude, which would be a win.
-	//  Maps and lists should also probably default off...?
-	//   (I have a feeling something might get touchy there.  Review when implementing those.)
-	//  Perhaps structs and unions are the only things likely to benefit from pointers.
-	return true
+
+	// As a simple heuristic,
+	// check how large the Go representation of this type will be.
+	// If it weighs little, we estimate that a pointer is not worthwhile,
+	// as storing the data directly will barely take more memory.
+	// Plus, the resulting code will be shorter and have fewer branches.
+
+	// The cutoff for "weighs little" is any size up to this number.
+	// It's hasn't been measured with any benchmarks or stats just yet.
+	// It's possible that, with those, it might increase in the future.
+	// Intuitively, any type 3x the size of a pointer is fine to inline.
+	// Adding a pointer will already add 1x overhead, anyway.
+	const pointerSizeSmallEnoughForInlining = 3
+
+	return pointerSizeOfSchemaType(t) < pointerSizeSmallEnoughForInlining
+}
+
+// pointerSizeOfSchemaType returns the size of a schema type,
+// relative to the size of a pointer in native Go.
+//
+// For example, TypeInt and TypeMap returns 1, but TypeList returns 3, as a
+// slice in Go has a pointer and two integers for length and capacity.
+// Any basic type smaller than a pointer, such as TypeBool, returns 1.
+func pointerSizeOfSchemaType(t schema.Type) int {
+	switch t.(type) {
+	// Go basic types which aren't bigger than a pointer.
+	// We assume that Enum is represented like Int.
+	case *schema.TypeBool, *schema.TypeInt, *schema.TypeFloat, *schema.TypeEnum:
+		return 1
+
+	// Go types which are just a pointer.
+	case *schema.TypeMap:
+		return 1
```

Comment:
I think this might technically be closer to 4, because our maps internally use both a golang map and a list to preserve order.

I'm not sure if that argues for bumping the threshhold to 4 or not.

Reply:
Oh, I wasn't aware. If it's indeed Go map + Go slice, then we should change it to 4.

Are these declared as Go types anywhere? Because if so, we could at least grab the sizes as globals, which would then be kept up to date as the types change.